### PR TITLE
generate proofs if none can be found

### PR DESCRIPTION
**Pull request by fbielejec**

### CODE REVIEW

**Review by deuszx — APPROVED** [review]
I'm not sure if this is the way to go -- i.e. I would prefer to inform the user about the missing keys and ask to generate separately but I think that for the time being we can go with this, more UX-friendly and simpler setup.

File: blender-cli/src/withdraw.rs [hunk 1]
```diff
@@ -86,6 +83,19 @@ pub(super) fn do_withdraw(
         recipient_bytes,
     );
 
+    let pk = match fs::read(proving_key_file) {
```
Comment (piotrMocz) [inline blender-cli/src/withdraw.rs:86]: I think you could extract it out into a utility parametrized by `deposit/withdraw` but it's a nitpick :)

File: blender-cli/src/deposit.rs [hunk 2]
```diff
@@ -20,18 +22,27 @@ pub(super) fn do_deposit(
...
     let circuit = DepositRelation::new(note, token_id, token_amount, trapdoor, nullifier);
+
+    let pk = match fs::read(proving_key_file) {
+        Ok(bytes) => bytes,
+        Err(_e) => {
+            let system = NonUniversalProvingSystem::Groth16;
+            let RawKeys { pk, vk } = system.generate_keys(circuit.clone());
+
+            fs::write("deposit.pk.bytes", pk.clone()).unwrap();
+            fs::write("deposit.vk.bytes", vk).unwrap();
```
Comment (pmikolajczyk41) [inline blender-cli/src/deposit.rs:40]: actually vk won't be of any use with this CLI

Comment (fbielejec) [reply blender-cli/src/deposit.rs:40]: it was to have it for registering in the pallet :)

**Pull request merged**